ovos_ww_plugin_openwakeword/__init__.py

```diff
@@ -48,12 +48,12 @@ def update(self, chunk):
         audio_frame = np.frombuffer(chunk, dtype=np.int16).tolist()
         self.audio_buffer.extend(audio_frame)  # build up the buffer until it has enough samples
 
-        if len(self.audio_buffer) >= 1280:
-            # Get prediction from openWakeWord
-            prediction = self.model.predict(self.audio_buffer)
+        while len(self.audio_buffer) >= 1280:
+            # Get prediction from openWakeWord.
+            prediction = self.model.predict(self.audio_buffer[:1280])
 
-            # Clear the buffer after each prediction
-            self.audio_buffer = []
+            # Clear the first 1280 bytes from buffer after each prediction
+            del self.audio_buffer[:1280]

```

Comment:
I’m not sure I knew `del` would operate sanely on a slice, but I’ve just verified that it does. Learn something new after 10 years. Thanks! 🥳